### ISSUE TITLE
[FEA] Polars engine CPU fallback for GPU out-of-memory errors

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#16835]
Today, users of the Polars GPU engine don't experience graceful performance degradation for GPU out-of-memory errors like users of cuDF pandas do -- instead, the query will fail. This is a pain point that's come up consistently in feedback from early users.

In parallel to investigating UVM, we should also investigate the feasibility of the current inspector-executor design supporting different types of CPU fallback when the GPU engine hits an out-of-memory error. E.g., we could rewind and execute the full query from scratch with the CPU engine or we could rewind only a portion of the query (which might require subgraph based hybrid execution).

**Comment by vyasr** [comment]
I think we're going to close this as a wontfix. Our existing in-memory engine now uses UVM, which will handle most cases, and the cases where it won't (e.g. a single column too large to fit in GPU memory) the performance of the entire query will be too bad to be worthwhile. The new streaming engine will solve this problem in a much more robust fashion.
